docs/pattern-matching.md

```diff
@@ -88,19 +88,23 @@ To ensure that recursive pattern matching functions don't loop in strict mode, i
 ```py
 # This is what the Foo function actually compiles to.
 # With -Olinearize-matches and -Ofloat-combinators (default on strict mode)
+# Main function
 (Foo) = λa λb λc (switch a { 0: Foo__C8; _: Foo__C9; } b c)
 
-(Foo__C5) = λa switch a { 0: λ* Baz; _: Foo__C4; }    # Foo.case_0
-(Foo__C4) = λ* λa (a Foo__C3)                         # Foo.case_0.case_false
-(Foo__C3) = λa switch a { 0: Baz; _: Foo__C2; }       # Part of cons pattern matching
-(Foo__C2) = λ* λa λb (b Foo__C1 a)                    # Foo.case_0.case_false.case_cons
-(Foo__C1) = λa switch a { 0: λ* Baz; _: Foo__C0; }    # Part of cons pattern matching
-(Foo__C0) = λ* λa λb λc (Bar c a b)                   # Foo.case_0.case_false.case_cons.case_cons
+# Case 0 branch
+(Foo__C8) = λa λb (a Foo__C5 b)                       # Foo.case_0.route
+(Foo__C5) = λa switch a { 0: λ* Baz; _: Foo__C4; }    # Foo.case_0.bool_true
+(Foo__C4) = λ* λa (a Foo__C3)                         # Foo.case_0.bool_false
+(Foo__C3) = λa switch a { 0: Baz; _: Foo__C2; }       # Foo.case_0.bool_false.case_cons1
+(Foo__C2) = λ* λa λb (b Foo__C1 a)                    # Foo.case_0.bool_false.case_cons1.handle
```

Comment:
Please use a consistent naming scheme for these cases, this is really confusing.

Either num_0/num_+ bool_false/bool_true etc
Or case_0/case_+ case_false/case_true etc.

What do you mean by route and handle?

```
0
0 true
0 false
0 false cons
0 false cons cons
0 false cons nil
0 false nil
+
+ false
+ true
```
these should be the cases.